akka-bench-jmh/src/main/scala/akka/actor/AffinityPoolComparativeBenchmark.scala

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (C) 2014-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.actor
+
+import java.util.concurrent.TimeUnit
+
+import akka.actor.BenchmarkActors._
+import akka.actor.ForkJoinActorBenchmark.cores
+import com.typesafe.config.ConfigFactory
+import org.openjdk.jmh.annotations._
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Array(Mode.Throughput))
+@Fork(1)
+@Threads(1)
+@Warmup(iterations = 10, time = 5, timeUnit = TimeUnit.SECONDS, batchSize = 1)
+@Measurement(iterations = 10, time = 15, timeUnit = TimeUnit.SECONDS, batchSize = 1)
+class AffinityPoolComparativeBenchmark {
```

Comment:
I tried this on my MacBook. I can't find any case where the affinity pool outperforms the FJP. When do we think it will be better?

Here is with `numThreads=8` and `numActors=16`:

```
[info] Benchmark                                           (dispatcher)                           (mailbox)  (throughPut)   Mode  Cnt         Score        Error  Units
[info] AffinityPoolComparativeBenchmark.pingPong    affinity-dispatcher  SingleConsumerOnlyUnboundedMailbox            25  thrpt   10  20341475.962 ± 830487.641  ops/s
[info] AffinityPoolComparativeBenchmark.pingPong  default-fj-dispatcher  SingleConsumerOnlyUnboundedMailbox            25  thrpt   10  22059251.037 ± 836438.018  ops/s
```

I suspect that the main bottleneck is elsewhere so I tried with an array based (allocation free) mailbox using Agrona's `ManyToOneConcurrentArrayQueue`. Better results:

```
[info] Benchmark                                           (dispatcher)  (mailbox)  (throughPut)   Mode  Cnt         Score         Error  Units
[info] AffinityPoolComparativeBenchmark.pingPong    affinity-dispatcher    default            25  thrpt   10  25638060.777 ± 2350585.697  ops/s
[info] AffinityPoolComparativeBenchmark.pingPong  default-fj-dispatcher    default            25  thrpt   10  30728843.419 ±  746318.152  ops/s
```